org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,513 @@
+
+    // @Nullable
+    private SecondGenerationConfiguration config;
+
+    @SuppressWarnings("unused")
+    // @Nullable
+    private SecondGenerationBindingConstants configurationConfig;
+
+    private final List<SecondGenerationChannelConfig> channelConfigs = new ArrayList<>();
+    private final List<SecondGenerationChannelConfig> channelConfigsExt = new ArrayList<>();
+    private final List<SecondGenerationChannelConfig> channelConfigsExtExt = new ArrayList<>();
+
+    public SecondGenerationHandler(Thing thing) {
+        super(thing);
+    }
+
+    // @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String url = config.url.toString();
+        String username = config.username;
+        String password = config.password;
+        String dxsIdConfiguration = "";
+        String valueConfiguration = "";
+
+        if (command instanceof RefreshType) {
+            logger.debug("Handle command for {} on channel {}: {}", thing.getUID(), channelUID, command);
+        }
+
+        if (command == RefreshType.REFRESH) {
+            return;
+        }
+
+        if (channelUID.getId().equals("chargeTimeEnd")) {
+            dxsIdConfiguration = "33556236";
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.configurationHandler(url, username, password, dxsIdConfiguration,
+                        valueConfiguration);
+            } catch (Exception e) {
+                logger.error("Unexpected error in KostalConfigurationHandler.configurationHandler, chargeTimeEnd !", e);
+            }
+        }
+
+        if (channelUID.getId().equals("batteryType")) {
+            dxsIdConfiguration = "33556252";
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.configurationHandler(url, username, password, dxsIdConfiguration,
+                        valueConfiguration);
+            } catch (Exception e) {
+                logger.error("Unexpected error in KostalConfigurationHandler.configurationHandler, batteryType !", e);
+            }
+        }
+
+        if (channelUID.getId().equals("batteryUsageConsumption")) {
+            dxsIdConfiguration = "33556249";
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.configurationHandler(url, username, password, dxsIdConfiguration,
+                        valueConfiguration);
+            } catch (Exception e) {
+                logger.error(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, batteryUsageConsumption  !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("batteryUsageStrategy")) {
+            dxsIdConfiguration = "83888896";
+            valueConfiguration = command.toString();
```

Comment:
Please just pass this data directly as parameters. Don't store them as local variables first, it serves no purpose other than to make the code more bloated.